api/document/js/Api.ts

```diff
@@ -0,0 +1,56 @@
+import axios from 'axios';
+
+import { getEl } from './util';
+
+function setStatus(msg: string, className: 'editor-status-error'|'' = '') {
+  const status = getEl('#status');
+  status.textContent = msg;
+  status.className = className;
+}
+
+function setStatusError(e: Error) {
+  let errMsg = `An error occurred: ${e}`;
+  const data = e['response'] && e['response']['data'];
+  if (data) {
+    errMsg += '\n' + JSON.stringify(data, null, 2);
+  }
+  setStatus(errMsg, 'editor-status-error');
+}
+
+export default class Api {
+  url: string;
+  contentType: string;
+  csrfToken: string;
+
+  constructor(url: string, contentType: string, csrfToken: string) {
```

Comment:
If it's possible to use destructuring notation with these arguments and have them passed-in as an object, that might make it easier to use and less easy to misuse, e.g. by confusing the ordering of `contentType` and `csrfToken`, especially as we move forward and potentially add more arguments to this constructor, but it's up to you!

Reply:
Good idea; I'll send that up.